src/de/dkfz/b080/co/common/BasicCOProjectsRuntimeService.groovy

```diff
@@ -40,13 +42,13 @@ public class BasicCOProjectsRuntimeService extends RuntimeService {
         def fs = context.getRuntimeService();
         //File cf = fs..createTemporaryConfigurationFile(executionContext);
         String pid = context.getDataSet().toString()
-        Map<String, Object> parameters = [
-                (COConstants.PRM_PID)            : (Object) pid,
-                (COConstants.PRM_PID_CAP)        : pid,
-                (COConstants.PRM_CONFIG_FILE)    : fs.getNameOfConfigurationFile(context).getAbsolutePath(),
-                (COConstants.PRM_ANALYSIS_DIR)   : context.getOutputDirectory().getParentFile().getParent()
+        Map<Object, Object> parameters = [
+                (COConstants.PRM_PID)         : (Object) pid,
+                (COConstants.PRM_PID_CAP)     : pid,
+                (COConstants.PRM_CONFIG_FILE) : fs.getNameOfConfigurationFile(context).getAbsolutePath(),
+                (COConstants.PRM_ANALYSIS_DIR): context.getOutputDirectory().getParentFile().getParent()
         ]
-        return parameters;
+        return parameters as Map<String, String>;
```

Comment:
Only a small thing, but  would this code not be clearer, if we used `(String) pid` and `getParent().toString()` and leave out the _post hoc_ type cast?

Reply:
Good point, I'll change it.